src/flint/test/test_all.py

```diff
@@ -2877,17 +2883,43 @@ def test_mpolys():
         assert raises(lambda: P(val={"bad": 1}, ctx=None), ValueError)
         assert raises(lambda: P(val="1", ctx=None), ValueError)
 
+        ctx1 = get_context(("x", 4))
+        ctx2 = get_context(("x", 4), ordering="deglex")
+        assert ctx1.drop_gens(*ctx1.names()).names() == tuple()
+        assert ctx1.drop_gens(ctx1.name(1), ctx1.name(2)).names() == (ctx1.name(0), ctx1.name(3))
+        assert ctx1.drop_gens().names() == ctx1.names()
+        assert ctx1.drop_gens(-1).names() == ctx1.names()[:-1]
+
+        assert ctx.infer_generator_mapping(ctx) == {i: i for i in range(ctx.nvars())}
+        assert ctx1.infer_generator_mapping(ctx) == {0: 0, 1: 1}
+        assert ctx1.drop_gens(*ctx.names()).infer_generator_mapping(ctx) == {}
+
+        # FIXME: Remove this guard when https://github.com/flintlib/flint/pull/2068 is
+        # resolved
+        if P is not flint.fmpz_mod_mpoly:
```

Comment:
As a temporary fix we could implement the conversion manually going via a dict.

Reply:
Good point.